sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -0,0 +1,307 @@
+        .setQueuePollTime(Read.DEFAULT_QUEUE_POLL_TIME)
+        .setQueueMaxPollTime(Read.DEFAULT_QUEUE_MAX_POLL_TIME)
+        .build();
+   }
+
+   /** Implementation of {@link #read}. */
+  @AutoValue
+  public abstract static class Read extends PTransform<PBegin, PCollection<String>> {
+    private static final long serialVersionUID = 2198301984784351829L;
+    public static final long DEFAULT_QUEUE_POLL_TIME = 50L;
+    public static final long DEFAULT_QUEUE_MAX_POLL_TIME = 3000L;
+
+    @Nullable abstract ValueProvider<String> getFilepattern();
+    @Nullable abstract ValueProvider<String> getTikaConfigPath();
+    @Nullable abstract Metadata getInputMetadata();
+    @Nullable abstract Boolean getReadOutputMetadata();
+    @Nullable abstract Long getQueuePollTime();
+    @Nullable abstract Long getQueueMaxPollTime();
+    @Nullable abstract Integer getMinimumTextLength();
+    @Nullable abstract Boolean getParseSynchronously();
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Builder setFilepattern(ValueProvider<String> filepattern);
+      abstract Builder setTikaConfigPath(ValueProvider<String> tikaConfigPath);
+      abstract Builder setInputMetadata(Metadata metadata);
+      abstract Builder setReadOutputMetadata(Boolean value);
+      abstract Builder setQueuePollTime(Long value);
+      abstract Builder setQueueMaxPollTime(Long value);
+      abstract Builder setMinimumTextLength(Integer value);
+      abstract Builder setParseSynchronously(Boolean value);
+
+      abstract Read build();
+    }
+
+    /**
+     * A {@link PTransform} that parses one or more files with the given filename
+     * or filename pattern and returns a bounded {@link PCollection} containing
+     * one element for each sequence of characters reported by Apache Tika SAX Parser.
+     *
+     * <p>Filepattern can be a local path (if running locally), or a Google Cloud Storage
+     * filename or filename pattern of the form {@code "gs://<bucket>/<filepath>"}
+     * (if running locally or using remote execution service).
+     *
+     * <p>Standard <a href="http://docs.oracle.com/javase/tutorial/essential/io/find.html" >Java
+     * Filesystem glob patterns</a> ("*", "?", "[..]") are supported.
+     */
+    public Read from(String filepattern) {
+      checkNotNull(filepattern, "Filepattern cannot be empty.");
+      return from(StaticValueProvider.of(filepattern));
+    }
+
+    /** Same as {@code from(filepattern)}, but accepting a {@link ValueProvider}. */
+    public Read from(ValueProvider<String> filepattern) {
+      checkNotNull(filepattern, "Filepattern cannot be empty.");
+      return toBuilder()
+          .setFilepattern(filepattern)
+          .setQueuePollTime(Read.DEFAULT_QUEUE_POLL_TIME)
+          .setQueueMaxPollTime(Read.DEFAULT_QUEUE_MAX_POLL_TIME)
+          .build();
+    }
+
+    /**
+     * Returns a new transform which will use the custom TikaConfig.
```

Comment:
Where can this config be located?

Reply:
It is a file resource location and can be provided with TikaOptions; more info at https://tika.apache.org/1.16/configuring.html